### PR TITLE
E2E storage: test snapshot semantics

### LINKED ISSUE #73625: E2E storage: test snapshot semantics
**What would you like to be added**:

https://github.com/kubernetes/kubernetes/pull/69036 adds a test that snapshots can be created and used. It would be good to actually verify that the snapshotting properly handles data (https://github.com/kubernetes/kubernetes/pull/69036/files#r252769254):

1. Write some data
1. Take a snapshot
1. Write more data
1. Create a new volume from snapshot
1. Validate data is the old data

**Why is this needed**:

To ensure that snapshotting really works.

/sig storage
/cc @msau42 @xing-yang @wackxu

### PR DESCRIPTION
**Pull request by zhucan**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
E2E storage: test snapshot semantics

**Which issue(s) this PR fixes**:

Fixes #73625

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
E2E storage: test snapshot semantics
```

### CODE REVIEW

**Comment by xing-yang** [comment]
Hi @zhucan, can you rebase this PR on top of this one: https://github.com/kubernetes/kubernetes/pull/80058.  Thanks.

**Comment by zhucan** [comment]
@xing-yang ok, I will do it later

**Comment by xing-yang** [comment]
@zhucan, please take a look of test failures.

File: test/e2e/storage/testsuites/snapshottable.go [hunk 1]
```diff
@@ -193,11 +221,69 @@ func (s *snapshottableTestSuite) defineTests(driver TestDriver, pattern testpatt
...
 		framework.ExpectEqual(volumeSnapshotRef["name"], snapshot.GetName())
 		framework.ExpectEqual(volumeSnapshotRef["namespace"], snapshot.GetNamespace())
+
+		ginkgo.By("creating a new claim from snapshot")
+		nPvc := e2epv.MakePersistentVolumeClaim(e2epv.PersistentVolumeClaimConfig{
+			ClaimSize:        claimSize,
+			StorageClassName: &(class.Name),
+		}, config.Framework.Namespace.Name)
+
+		//nPvc.Spec.StorageClassName = &class.Name
```
Comment (xing-yang) [inline test/e2e/storage/testsuites/snapshottable.go]: Remove the commented out line.

**Comment by xing-yang** [comment]
@zhucan My PR is already merged.  Can you rebase against the master?

**Comment by zhucan** [comment]
@xing-yang I will rebase it out of day

**Comment by xing-yang** [comment]
@zhucan can you please rebase this PR.

**Comment by xing-yang** [comment]
Note that my PR is already merged.  After rebasing from master, we should not see my PR in this PR any more.

**Comment by zhucan** [comment]
@xing-yang copy that

**Comment by xing-yang** [comment]
@zhucan Can you rebase this PR?  Thanks.

File: cluster/addons/volumesnapshot-crds/OWNERS [hunk 2]
```diff
@@ -0,0 +1,7 @@
+approvers:
```
Comment (xing-yang) [inline cluster/addons/volumesnapshot-crds/OWNERS:1]: These changes should not show up as new changes in this PR.  Please resolve the rebase issues.

**Pull request closed without merging**